framework/src/executioners/Executioner.C

```diff
@@ -103,3 +110,69 @@ Executioner::addAttributeReporter(const std::string & name, Real initial_value)
 
   return value;
 }
+
+void
+Executioner::addSolveObject(const std::string & type, const std::string & name,
+                            InputParameters & parameters)
+{
+  if (_solve_object_names.count(type))
+    mooseError("Solve object in type ", type, " has been constructed");
```

Comment:
what if there's more than one solve of the same type?

Reply:
I think we should be a little conservative before making this executioner really fancy. We have not convert AMR, timestepping, etc. Having multiple solve objects in the same type does not make much sense to me at this point. Another point is to have solve object to set multiple inner solve objects, which in my view we should pursue a little slowly.

Reply:
I want steady, transient and eigenvalue to be able to use the solve object nesting, and I dont want an alternative general execute unless we can prove that we need this.

Reply:
Maybe you are replying the other comment. To make sure, you want `Executioner::execute` to be removed, right?

Reply:
I agree with that, I dont think the need has been expressed for multiple solve objects, eg a tree, whereas the need is clear for a linear chain of solve objects

Reply:
I can move this code into `test`. We need this to test the solve object in input syntax anyway. When ready, we can promote the code into framework.

Reply:
ok I think the comments are mixed up. Are you replying to this thread or the other one?
eg are you talking about moving execute or this check on the object names?

Reply:
I am having hard time on following comments... The previous comment is for the other one.